pallet/src/lib.rs

```diff
@@ -423,4 +439,20 @@ pub mod pallet {
             Ok(validity)
         }
     }
+
+    impl<T: Config> Pallet<T> {
+        /// NOTE this is manually configured based on attributes of Attribute struct,
+        /// was Attribute struct to change in the future, this function would be modified also
+        pub fn deposit_amount() -> BalanceOf<T> {
+            // see pub struct Attribute
+            let attribute_size = (T::BoundedDataLen::get() * 2) as usize
+                + T::AccountId::max_encoded_len()
+                + TimeOf::<T>::max_encoded_len();
+
+            // amount for the storage deposit
+            let deposit = T::StorageDepositBase::get()
```

Comment:
Should we use the saturate_mul?

Reply:
Considering the types in question are generic over the runtime config, i think we should. Good idea, i'll have a look at these.